### ISSUE TITLE
"multiple return types for content-type ..." problem not shown in VS Code

### ISSUE THREAD

**Issue by mikekistler** [microsoft/cadl#1419]
Removing the `@error` decorator from the error schema will produce a compile error:
```
error @cadl-lang/rest/duplicate-response: Multiple return types for content type application/json and status code 200
```

But this error is not highlighted when the code is viewed in VS Code with the Cadl VS Code extension enabled.

<img width="1077" alt="image" src="https://user-images.githubusercontent.com/85643503/206864276-2f8cfeab-3da5-4781-9dd2-eceb6ed7abeb.png">

Compiler version: Cadl compiler v0.38.3

**Comment by timotheeguerin** [comment]
Ok so we found that the error is due to the `auth.cadl` has an invalid js doc by having `p@55w0rd ` string inside

```cadl
/**
 *  p@55w0rd 
 */
model BasicAuth {}
```
but because right now only the ide parse jsdoc then we odn't see that issue in the compiler.

The reason it doesn't show anythjing is :
1. we don't show errors in non opened files: in this case an insalled library file
2. if there is any errors we don't continue to the next step(validate) to prevent cascade of errors

**Comment by nguerrera** [comment]
Immediate issue fixed by #1438 (#1442 merged to main)

Follow-up issues to prevent regression:

#1430, #1441
